### ISSUE TITLE
`InstanceNorm1d` do not support 2d input

### ISSUE THREAD

**Issue by innerlee** [pytorch/pytorch#4170]
Ref [doc](http://pytorch.org/docs/master/nn.html#torch.nn.InstanceNorm1d), the input only supports `(N, C, L)` format. This is not consistent with [BatchNorm1d](http://pytorch.org/docs/master/nn.html#torch.nn.BatchNorm1d) which supports also `(N, C)`.

**Comment by soumith** [comment]
@innerlee it doesn't make much sense to support instance norm for `(N, C)` because it is usually normalizing over `L` dimension.

**Comment by innerlee** [comment]
Okay, so the documentation should be updated?
